react-components/src/architecture/concrete/annotation360/createTriangleIndexesFromVectors.test.ts

```diff
@@ -0,0 +1,124 @@
+    const vectors: Vector3[] = [];
+    for (let i = 0; i < 10; i++) {
+      vectors.push(new Vector3(i, i, i));
+      const indexes = createTriangleIndexesFromVectors(vectors);
+      expect(indexes).toBeUndefined();
+    }
+  });
+
+  test('should return undefined for less than 3 vectors', () => {
+    const vectors = [new Vector3(1, 0, 0), new Vector3(0, 1, 0)];
+    const indexes = createTriangleIndexesFromVectors(vectors);
+    expect(indexes).toBeUndefined();
+  });
+
+  test('should return undefined for collinear vectors', () => {
+    const vectors = [new Vector3(0, 0, 0), new Vector3(1, 1, 1), new Vector3(2, 2, 2)];
+    const indexes = createTriangleIndexesFromVectors(vectors);
+    expect(indexes).toBeUndefined();
+  });
+
+  test('should return correct indexes for a simple triangle', () => {
+    const vectors = [new Vector3(0, 0, -1), new Vector3(1, 0, -1), new Vector3(0, 1, -1)];
+    const indexes = createTriangleIndexesFromVectors(vectors);
+    expect(indexes).toEqual([1, 2, 0]);
+  });
+
+  test('should handle vectors forming a square', () => {
+    const vectors = createSquareToEdgeOfUnitSphere();
+    const indexes = createTriangleIndexesFromVectors(vectors);
+    expect(indexes).toEqual([2, 3, 0, 0, 1, 2]);
+  });
+
+  test('should handle reversed vectors forming a square', () => {
+    const vectors = createSquareToEdgeOfUnitSphere();
+    vectors.reverse();
+    const indexes = createTriangleIndexesFromVectors(vectors);
+    expect(indexes).toEqual([1, 0, 3, 3, 2, 1]);
+  });
+
+  test('should handle colinear vectors in a square', () => {
+    const vectors = createSquareToEdgeOfUnitSphere(true);
+    const indexes = createTriangleIndexesFromVectors(vectors);
+    expect(indexes).toEqual([4, 0, 2, 2, 3, 4]);
+  });
+
+  test('should handle vectors forming a pentagon', () => {
+    const vectors = createPentagonToEdgeOfUnitSphere();
+    const indexes = createTriangleIndexesFromVectors(vectors);
+    expect(indexes).toEqual([4, 0, 1, 2, 3, 4, 4, 1, 2]);
+  });
+
+  test('should handle reversed vectors forming a pentagon', () => {
+    const vectors = createPentagonToEdgeOfUnitSphere();
+    vectors.reverse();
+    const indexes = createTriangleIndexesFromVectors(vectors);
+    expect(indexes).toEqual([0, 4, 3, 2, 1, 0, 0, 3, 2]);
+  });
+});
+
+/**
+ * Creates a square of vectors positioned on the edge of a unit sphere.
+ *
+ * The function generates four 3D vectors representing the vertices of a square
+ * in the XY plane at a fixed Z-coordinate. Each vector is normalized to lie on
+ * the surface of a unit sphere.
+ *
+ */
+
+function createSquareToEdgeOfUnitSphere(addOnColinearPoint = false): Vector3[] {
+  const vectors: Vector3[] = [];
+  const z = -5;
+  vectors.push(new Vector3(0, 0, z));
+  vectors.push(new Vector3(1, 0, z));
+  vectors.push(new Vector3(1, 1, z));
+  vectors.push(new Vector3(0, 1, z));
+
+  forEach(vectors, (vector) => vector.normalize());
```

Comment:
Just to remove some "any" lint warnings

```suggestion
  forEach(vectors, (vector: Vector3) => vector.normalize());
```